docs/config.md

```diff
@@ -0,0 +1,156 @@
+              - ^web-.+$
+      - matchExpressions:
+          - key: hostname
+            operator: Regexp
+            values:
+              - ^web-.+$
+
+    # Add/replace annotations and labels for nodes that match the transformation
+    annotations:
+      # You can use the Go template to get the value of the platform metadata variable
+      talos.dev/instance-id: "id-{{ .InstanceID }}"
+    labels:
+      # Add label to the node, in this case, we add node role label
+      node-role.kubernetes.io/web: ""
+
+    # Replace platform metadata variables for nodes that match the transformation
+    platformMetadata:
+      Region: "{{ .Region }}-on-metal"
+      Zone: "us-west-1f"
+```
+
+### Transformations parameters:
+
+* `nodeSelector` - a list of node selector requirements by platfrom metadata variable.
+  * `matchExpressions` - a list of node selector requirements by platfrom metadata variable.
+    * `key` - the key that the selector applies to, case `insensitive`.
+    * `operator` - represents a key's relationship to a set of values. Supported operators are `In`, `NotIn`, `Exists`, `DoesNotExist`, `Gt`, `Lt`, `Regexp`.
+    * `values` - an array of string values.
+
+* `annotations` - a map of key-value pairs to add to each node that matches the transformation.
+  * `key` - the key of the annotation.
+  * `value` - the value of the annotation. You can use the [Go template](https://golang.org/pkg/text/template/) to get the value of the platform metadata variable. Variables are case `sensitive`.
+
+* `labels` - a map of key-value pairs to add to each node that matches the transformation.
+  * `key` - the key of the label.
+  * `value` - the value of the label. You can use the [Go template](https://golang.org/pkg/text/template/) to get the value of the platform metadata variable. Variables are case `sensitive`.
+
+* `platformMetadata` - a map of key-value pairs to add to each node that matches the transformation.
+  * `key` - the key of the platform metadata variable to replace.
+  * `value` - the value of the platform metadata variable. You can use the [Go template](https://golang.org/pkg/text/template/) to get the value of the platform metadata variable. Variables are case `sensitive`.
+
+### Platform metadata variables:
+
+Go struct for platform metadata,
+original code: [platform_metadata.go](https://github.com/siderolabs/talos/blob/main/pkg/machinery/resources/runtime/platform_metadata.go)
+
+```go
+type PlatformMetadataSpec struct {
+	Platform     string `yaml:"platform,omitempty" protobuf:"1"`
+	Hostname     string `yaml:"hostname,omitempty" protobuf:"2"`
+	Region       string `yaml:"region,omitempty" protobuf:"3"`
+	Zone         string `yaml:"zone,omitempty" protobuf:"4"`
+	InstanceType string `yaml:"instanceType,omitempty" protobuf:"5"`
+	InstanceID   string `yaml:"instanceId,omitempty" protobuf:"6"`
+	ProviderID   string `yaml:"providerId,omitempty" protobuf:"7"`
+	Spot         bool   `yaml:"spot,omitempty" protobuf:"8"`
```

Comment:
In AWS this is called "provisioning type" or "provisioning model" (GCP) and would be a string like `spot`, `on-demand`, or `reserved`.

Reply:
Spot, if true - can terminate at any time (cloud decision).  on-demand and reserved - your decision

Reply:
Is the goal of this field to set the controller to do specific things with spot instances only?

Reply:
Do not deploy pods which cannot migrate fast (long task jobs for instance)

Reply:
If our goal is to help users not run jobs that cannot terminate quickly I think spot: true/false is to vague. There are other reasons they may want to target spot, on-demand, and reserved instance types. And not every cloud provider calls spare capacity "spot".

Karpenter uses the [`capacity-type`](https://karpenter.sh/v0.32/concepts/scheduling/#well-known-labels) label which I think would be a more generic way to present this information that works across different cloud environments.

Reply:
OK, no problem. As long as we can change it it the future if we need to.